Dockerfile

```
@@ -49,17 +47,30 @@ RUN mkdir -p /etc/nix && \
+RUN --mount=type=secret,id=github_token,mode=0444 \
+    set -e; \
+    /usr/bin/nix-daemon & \
+    sleep 5 && \
+    sudo -u "$USER" -H env GITHUB_TOKEN_FILE=/run/secrets/github_token IN_DOCKER=true USER="$USER" \
+      make -C /tmp/dotfiles-cache nix-cache-warmup
+
+ARG COMMIT_SHA=main
+ARG GITHUB_PR=
 
-RUN /usr/bin/nix-daemon & \
+RUN --mount=type=secret,id=github_token,mode=0444 \
```

Hadolint (2.14.0)

[error] Do not use sudo as it leads to unpredictable behavior. Use a tool like gosu to enforce root

(DL3004)